### ISSUE TITLE
AttributeError: 'Name' object has no attribute 'expr'

### ISSUE THREAD

**Issue by MichaelCook** [PyCQA/pylint#6828]
### Bug description

When parsing the following file:

```python
#!/usr/bin/env python3
import sys
import argparse
import logging
import socket
import select
import hashlib
import time
import subprocess
from typing import Final
from pathlib import Path

# ----------------------------------------------------------------------------

THIS_SCRIPT: Final = Path(sys.argv[0]).stem

def parse_args() -> argparse.Namespace:
    parser = argparse.ArgumentParser(description="""
Talk to the Podcast app via a TCP connection
""")

    parser.add_argument('--phone', default=[], metavar='H', action='append',
                        help="""
The host names or IP addresses of the phone.
Comma-separated, and may be given multiple times
""")

    parser.add_argument('--keep-alive', '-k', default=30.0, metavar='SEC', type=float,
                        help="""
If we haven't received anything from the phone in this amount of time,
send a message to check the connection (default: %(default)r)
""")

    parser.add_argument('--debug', action='store_true')

    parser.add_argument('commands', nargs='*', help="""

Without arguments, go interactive.  With arguments, send each argument as a
command and wait for an OK or ERR response.
In Bash, use the $'...\\t...' construct to add tabs to the commands

""")

    return parser.parse_args()

OPTS: Final = parse_args()

if OPTS.commands:
    logging.basicConfig(level=logging.DEBUG if OPTS.debug else logging.INFO,
                        format='%(name)s: %(levelname)s: %(message)s')
else:
    logging.basicConfig(level=logging.DEBUG if OPTS.debug else logging.INFO,
                        format='%(asctime)s [%(levelname).1s] %(message)s')
LOGGER: Final = logging.getLogger(THIS_SCRIPT)

# ----------------------------------------------------------------------------

PHONE_PORT: Final = 4004

class TcpClient:

    def __init__(self, addresses: list[str]) -> None:
        self.addresses = addresses
        self.socket: socket.socket | None = None
        self.blocking = True
        self.buffer = b''
        self.remote_disconnected = False

    def rotate(self) -> None:
        self.addresses = self.addresses[1:] + [self.addresses[0]]

    def connect(self) -> bool:
        if self.socket:
            self.socket.close()
            self.socket = None
        self.socket = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
        self.socket.setsockopt(socket.IPPROTO_TCP, socket.TCP_NODELAY, 1)
        self.socket.settimeout(5)

        phone = self.addresses[0], PHONE_PORT
        LOGGER.debug('Connecting to %s:%d', phone[0], phone[1])
        try:
            self.socket.connect(phone)
            return True
        except ConnectionRefusedError:
            LOGGER.debug('Connection refused')
        except OSError as exc:
            LOGGER.debug('OSError: %s', exc.strerror)
        self.rotate()
        return False

    def readline(self, blocking: bool) -> bytes | None:
        """
        Read a line from the socket.  If a complete line is available, return the
        line including the '\n'.  Otherwise, return ''.  But if the remote has
        disconnected, return a remaining partial line if any, otherwise return None
        """
        self.setblocking(blocking)
        i = self.buffer.find(b'\n')
        if i == -1:
            self.fill()
            i = self.buffer.find(b'\n')
            if i == -1:
                if self.remote_disconnected:
                    if not self.buffer:
                        return None
                    line = self.buffer
                    self.buffer = b''
                    return line
                return b''
        line = self.buffer[:i+1]
        self.buffer = self.buffer[i+1:]
        return line

    def fill(self) -> None:
        """
        Read whatever is available from the socket and append to the buffer
        """
        try:
            while True:
                LOGGER.debug('reading socket...')
                assert self.socket is not None
                read = self.socket.recv(4096)
                LOGGER.debug('read [%d] %r', len(read), read)
                if not read:
                    self.remote_disconnected = True
                    break
                self.buffer += read
                if b'\n' in read:
                    break
        except BlockingIOError:
            pass
        except TimeoutError:
            LOGGER.error('Timeout')
            sys.exit(1)
        except ConnectionResetError as exc:
            LOGGER.error('%s', exc.strerror)
            sys.exit(1)
        LOGGER.debug('fill: [%d] %r', len(self.buffer), self.buffer)

    def setblocking(self, blocking: bool = True) -> None:
        assert self.socket is not None
        if blocking != self.blocking:
            self.socket.setblocking(blocking)
            self.blocking = blocking

    def writeline(self, line: bytes | str) -> None:
        assert self.socket is not None
        if isinstance(line, str):
            line = line.encode('utf-8')
        self.setblocking(True)
        if not line.endswith(b'\n'):
            line += b'\n'
        self.socket.send(line)

def md5(data: bytes) -> str:
    return hashlib.md5(data).hexdigest()

def authenticate(line: bytes, sock: TcpClient) -> None:
    password = Path('~/.local/pod-app').expanduser().read_text(encoding='utf-8').strip()
    hello = b'HELLO\t'
    if not line.startswith(hello):
        raise Exception(f'Wrong greeting {line!r}')
    nonce = line[len(hello):].decode('utf-8').strip()
    LOGGER.debug('nonce=%r', nonce)
    msg = bytes.fromhex(nonce) + password.encode('utf-8')
    sock.writeline(hello + md5(msg).encode('utf-8'))

def do_interactive(addresses: list[str]) -> None:
    sock = TcpClient(addresses)

    LOGGER.debug('Connecting')
    i = 0
    while not sock.connect():
        i += 1
        if i == 4:
            LOGGER.warning('Connecting')
        time.sleep(.5)
    if i >= 4:
        LOGGER.info('Connected')

    while True:
        line = sock.readline(blocking=True)
        if not line:
            LOGGER.error('Server disconnected')
            sys.exit(1)
        if not line.startswith(b'NFY\t'):
            break
        LOGGER.warning('Discard unexpected response: %r', line)
    authenticate(line, sock)

    LOGGER.debug('Interactive')
    fds = [sock.socket, sys.stdin]
    while True:
        readable, _, _ = select.select(fds, [], [], OPTS.keep_alive)
        LOGGER.debug('Readable (select) [%d] %r', len(readable), readable)

        if not readable:
            LOGGER.debug('Timeout')
            sock.writeline('')
            continue

        if sock.socket in readable:
            LOGGER.debug('socket is readable')
            while line := sock.readline(blocking=False):
                LOGGER.debug('from server: %r', line)
                LOGGER.info('%s', line.decode('utf-8').rstrip('\r\n').replace('\t', '⋮ '))
            if line is None:
                break

        if sys.stdin in readable:
            LOGGER.debug('stdin is readable')
            line = sys.stdin.buffer.readline()
            if not line:
                break
            LOGGER.debug('from user: %r', line)
            sock.writeline(line)

def do_command(sock: TcpClient, command: str) -> bool:
    sock.writeline(command)

    while line_bytes := sock.readline(blocking=True):
        line = line_bytes.decode('utf-8').rstrip('\r\n')
        fields = line.split('\t', maxsplit=1)
        if fields[0] == 'ERR':
            LOGGER.error('%s', line)
            return False
        print(line)
        if fields[0] == 'OK':
            return True
    LOGGER.error('No OK or ERR')
    return False

def do_commands(addresses: list[str], commands: list[str]) -> None:
    sock = TcpClient(addresses)
    for _ in range(2 * len(addresses)):  # try each address twice
        if sock.connect():
            break
    else:
        LOGGER.critical('Failed to connect to %s', addresses)
        sys.exit(1)

    line = sock.readline(blocking=True)
    if not line:
        LOGGER.error('Server disconnected')
        sys.exit(1)
    authenticate(line, sock)

    trouble = False
    for command in commands:
        if not do_command(sock, command):
            trouble = True
    if trouble:
        sys.exit(1)

def scanport() -> str:
    cmd = ['scanport', '--timeout', '10', '--quiet', '--port', str(PHONE_PORT)]
    LOGGER.debug('Run: %s', ' '.join(cmd))
    with subprocess.Popen(cmd, stdout=subprocess.PIPE) as ph:
        assert ph.stdout is not None
        for line_bytes in iter(ph.stdout.readline, b''):
            line = line_bytes.decode('utf-8', 'backslashreplace').rstrip('\r\n')
            address, _port = line.split(':', maxsplit=1)
            LOGGER.debug('Found phone at %r', address)
            ph.kill()
            ph.wait()
            return address
        ph.wait()
    raise Exception('Phone not found')

def main() -> None:
    # Convert OPTS.phone from a list of comma-separated names to a flat list
    #   OPTS.phone = ['one,two,three', 'four', 'five,six']
    # becomes
    #   ['one', 'two', 'three', 'four', 'five', 'six']
    addresses: list[str]
    if OPTS.phone:
        addresses = sum([i.split(',') for i in OPTS.phone], [])
    else:
        addresses = [scanport()]
    LOGGER.debug('addresses %r', addresses)

    if OPTS.commands:
        do_commands(addresses, OPTS.commands)
    else:
        do_interactive(addresses)

main()

```

pylint crashed with a ``AstroidError`` and with the following stacktrace:
```
Traceback (most recent call last):
  File "/usr/local/lib/python3.10/dist-packages/pylint/lint/pylinter.py", line 713, in _check_file
    check_astroid_module(ast_node)
  File "/usr/local/lib/python3.10/dist-packages/pylint/lint/pylinter.py", line 912, in check_astroid_module
    retval = self._check_astroid_module(
  File "/usr/local/lib/python3.10/dist-packages/pylint/lint/pylinter.py", line 962, in _check_astroid_module
    walker.walk(node)
  File "/usr/local/lib/python3.10/dist-packages/pylint/utils/ast_walker.py", line 93, in walk
    self.walk(child)
  File "/usr/local/lib/python3.10/dist-packages/pylint/utils/ast_walker.py", line 90, in walk
    callback(astroid)
  File "/usr/local/lib/python3.10/dist-packages/pylint/extensions/private_import.py", line 69, in visit_importfrom
    private_names = self._get_type_annotation_names(node, names)
  File "/usr/local/lib/python3.10/dist-packages/pylint/extensions/private_import.py", line 121, in _get_type_annotation_names
    self._populate_type_annotations(node.root(), self.all_used_type_annotations)
  File "/usr/local/lib/python3.10/dist-packages/pylint/extensions/private_import.py", line 170, in _populate_type_annotations
    ] = self._assignments_call_private_name(name_assignments, private_name)
  File "/usr/local/lib/python3.10/dist-packages/pylint/extensions/private_import.py", line 237, in _assignments_call_private_name
    current_attribute = current_attribute.expr
AttributeError: 'Name' object has no attribute 'expr'

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/usr/local/lib/python3.10/dist-packages/pylint/lint/pylinter.py", line 670, in _check_files
    self._check_file(get_ast, check_astroid_module, file)
  File "/usr/local/lib/python3.10/dist-packages/pylint/lint/pylinter.py", line 715, in _check_file
    raise astroid.AstroidError from e
astroid.exceptions.AstroidError
```

### Configuration

```ini
[MAIN]

# Analyse import fallback blocks. This can be used to support both Python 2 and
# 3 compatible code, which means that the block might have code that exists
# only in one or another interpreter, leading to false positives when analysed.
analyse-fallback-blocks=no

# Load and enable all available extensions. Use --list-extensions to see a list
# all available extensions.
#enable-all-extensions=

# In error mode, checkers without error messages are disabled and for others,
# only the ERROR messages are displayed, and no reports are done by default.
#errors-only=

# Always return a 0 (non-error) status code, even if lint errors are found.
# This is primarily useful in continuous integration scripts.
#exit-zero=

# A comma-separated list of package or module names from where C extensions may
# be loaded. Extensions are loading into the active Python interpreter and may
# run arbitrary code.
extension-pkg-allow-list=

# A comma-separated list of package or module names from where C extensions may
# be loaded. Extensions are loading into the active Python interpreter and may
# run arbitrary code. (This is an alternative name to extension-pkg-allow-list
# for backward compatibility.)
extension-pkg-whitelist=PyQt5

# Return non-zero exit code if any of these messages/categories are detected,
# even if score is above --fail-under value. Syntax same as enable. Messages
# specified are enabled, while categories only check already-enabled messages.
fail-on=

# Specify a score threshold to be exceeded before program exits with error.
fail-under=10.0

# Interpret the stdin as a python script, whose filename needs to be passed as
# the module_or_package argument.
#from-stdin=

# Files or directories to be skipped. They should be base names, not paths.
ignore=CVS

# Add files or directories matching the regex patterns to the ignore-list. The
# regex matches against paths and can be in Posix or Windows format.
ignore-paths=

# Files or directories matching the regex patterns are skipped. The regex
# matches against base names, not paths. The default value ignores Emacs file
# locks
ignore-patterns=

# List of module names for which member attributes should not be checked
# (useful for modules/projects where namespaces are manipulated during runtime
# and thus existing member attributes cannot be deduced by static analysis). It
# supports qualified module names, as well as Unix pattern matching.
ignored-modules=

# Python code to execute, usually for sys.path manipulation such as
# pygtk.require().
#init-hook=

# Use multiple processes to speed up Pylint. Specifying 0 will auto-detect the
# number of processors available to use.
jobs=1

# Control the amount of potential inferred values when inferring a single
# object. This can help the performance when dealing with large functions or
# complex, nested conditions.
limit-inference-results=100

# List of plugins (as comma separated values of python module names) to load,
# usually to register additional checkers.
load-plugins=

# Pickle collected data for later comparisons.
persistent=yes

# Minimum Python version to use for version dependent checks. Will default to
# the version used to run pylint.
py-version=3.10

# Discover python modules and packages in the file system subtree.
recursive=no

# When enabled, pylint would attempt to guess common misconfiguration and emit
# user-friendly hints instead of false-positive error messages.
suggestion-mode=yes

# Allow loading of arbitrary C extensions. Extensions are imported into the
# active Python interpreter and may run arbitrary code.
unsafe-load-any-extension=no

# In verbose mode, extra non-checker-related info will be displayed.
#verbose=


[REPORTS]

# Python expression which should return a score less than or equal to 10. You
# have access to the variables 'fatal', 'error', 'warning', 'refactor',
# 'convention', and 'info' which contain the number of messages in each
# category, as well as 'statement' which is the total number of statements
# analyzed. This score is used by the global evaluation report (RP0004).
evaluation=10.0 - ((float(5 * error + warning + refactor + convention) / statement) * 10)

# Template used to display messages. This is a python new-style format string
# used to format the message information. See doc for all details.
msg-template=

# Set the output format. Available formats are text, parseable, colorized, json
# and msvs (visual studio). You can also give a reporter class, e.g.
# mypackage.mymodule.MyReporterClass.
#output-format=

# Tells whether to display a full report or only the messages.
reports=no

# Activate the evaluation score.
score=yes


[MESSAGES CONTROL]

# Only show warnings with the listed confidence levels. Leave empty to show
# all. Valid levels: HIGH, CONTROL_FLOW, INFERENCE, INFERENCE_FAILURE,
# UNDEFINED.
confidence=HIGH,
           CONTROL_FLOW,
           INFERENCE,
           INFERENCE_FAILURE,
           UNDEFINED

# Disable the message, report, category or checker with the given id(s). You
# can either give multiple identifiers separated by comma (,) or put this
# option multiple times (only on the command line, not in the configuration
# file where it should appear only once). You can also use "--disable=all" to
# disable everything first and then re-enable specific checks. For example, if
# you want to run only the similarities checker, you can use "--disable=all
# --enable=similarities". If you want to run only the classes checker, but have
# no Warning level messages displayed, use "--disable=all --enable=classes
# --disable=W".
disable=compare-to-empty-string,
        compare-to-zero,
        consider-ternary-expression,
        consider-using-assignment-expr,
        empty-comment,
        docstring-first-line-empty,
        global-statement,
        invalid-name,
        missing-class-docstring,
        missing-function-docstring,
        missing-module-docstring,
        redefined-loop-name,
        too-complex,
        too-few-public-methods,
        too-many-arguments,
        too-many-branches,
        too-many-instance-attributes,
        too-many-lines,
        too-many-locals,
        too-many-return-statements,
        too-many-statements,
        too-many-try-statements,
        while-used,

# Enable the message, report, category or checker with the given id(s). You can
# either give multiple identifier separated by comma (,) or put this option
# multiple time (only on the command line, not in the configuration file where
# it should appear only once). See also the "--disable" option for examples.
enable=c-extension-no-member


[LOGGING]

# The type of string formatting that logging methods do. `old` means using %
# formatting, `new` is for `{}` formatting.
logging-format-style=old

# Logging modules to check that the string format arguments are in logging
# function parameter format.
logging-modules=logging


[MISCELLANEOUS]

# List of note tags to take in consideration, separated by a comma.
notes=FIXME,
      XXX,
      TODO

# Regular expression of note tags to take in consideration.
notes-rgx=


[BASIC]

# Naming style matching correct argument names.
argument-naming-style=snake_case

# Regular expression matching correct argument names. Overrides argument-
# naming-style. If left empty, argument names will be checked with the set
# naming style.
#argument-rgx=

# Naming style matching correct attribute names.
attr-naming-style=snake_case

# Regular expression matching correct attribute names. Overrides attr-naming-
# style. If left empty, attribute names will be checked with the set naming
# style.
#attr-rgx=

# Bad variable names which should always be refused, separated by a comma.
bad-names=foo,
          bar,
          baz,
          toto,
          tutu,
          tata

# Bad variable names regexes, separated by a comma. If names match any regex,
# they will always be refused
bad-names-rgxs=

# Naming style matching correct class attribute names.
class-attribute-naming-style=any

# Regular expression matching correct class attribute names. Overrides class-
# attribute-naming-style. If left empty, class attribute names will be checked
# with the set naming style.
#class-attribute-rgx=

# Naming style matching correct class constant names.
class-const-naming-style=UPPER_CASE

# Regular expression matching correct class constant names. Overrides class-
# const-naming-style. If left empty, class constant names will be checked with
# the set naming style.
#class-const-rgx=

# Naming style matching correct class names.
class-naming-style=PascalCase

# Regular expression matching correct class names. Overrides class-naming-
# style. If left empty, class names will be checked with the set naming style.
#class-rgx=

# Naming style matching correct constant names.
const-naming-style=UPPER_CASE

# Regular expression matching correct constant names. Overrides const-naming-
# style. If left empty, constant names will be checked with the set naming
# style.
#const-rgx=

# Minimum line length for functions/classes that require docstrings, shorter
# ones are exempt.
docstring-min-length=-1

# Naming style matching correct function names.
function-naming-style=snake_case

# Regular expression matching correct function names. Overrides function-
# naming-style. If left empty, function names will be checked with the set
# naming style.
#function-rgx=

# Good variable names which should always be accepted, separated by a comma.
good-names=i,
           j,
           k,
           ex,
           Run,
           _

# Good variable names regexes, separated by a comma. If names match any regex,
# they will always be accepted
good-names-rgxs=

# Include a hint for the correct naming format with invalid-name.
include-naming-hint=no

# Naming style matching correct inline iteration names.
inlinevar-naming-style=any

# Regular expression matching correct inline iteration names. Overrides
# inlinevar-naming-style. If left empty, inline iteration names will be checked
# with the set naming style.
#inlinevar-rgx=

# Naming style matching correct method names.
method-naming-style=snake_case

# Regular expression matching correct method names. Overrides method-naming-
# style. If left empty, method names will be checked with the set naming style.
#method-rgx=

# Naming style matching correct module names.
module-naming-style=snake_case

# Regular expression matching correct module names. Overrides module-naming-
# style. If left empty, module names will be checked with the set naming style.
#module-rgx=

# Colon-delimited sets of names that determine each other's naming style when
# the name regexes allow several styles.
name-group=

# Regular expression which should only match function or class names that do
# not require a docstring.
no-docstring-rgx=^_

# List of decorators that produce properties, such as abc.abstractproperty. Add
# to this list to register other decorators that produce valid properties.
# These decorators are taken in consideration only for invalid-name.
property-classes=abc.abstractproperty

# Regular expression matching correct type variable names. If left empty, type
# variable names will be checked with the set naming style.
#typevar-rgx=

# Naming style matching correct variable names.
variable-naming-style=snake_case

# Regular expression matching correct variable names. Overrides variable-
# naming-style. If left empty, variable names will be checked with the set
# naming style.
#variable-rgx=


[IMPORTS]

# List of modules that can be imported at any level, not just the top level
# one.
allow-any-import-level=

# Allow wildcard imports from modules that define __all__.
allow-wildcard-with-all=no

# Deprecated modules which should not be used, separated by a comma.
deprecated-modules=optparse,tkinter.tix

# Output a graph (.gv or any supported image format) of external dependencies
# to the given file (report RP0402 must not be disabled).
ext-import-graph=

# Output a graph (.gv or any supported image format) of all (i.e. internal and
# external) dependencies to the given file (report RP0402 must not be
# disabled).
import-graph=

# Output a graph (.gv or any supported image format) of internal dependencies
# to the given file (report RP0402 must not be disabled).
int-import-graph=

# Force import order to recognize a module as part of the standard
# compatibility libraries.
known-standard-library=

# Force import order to recognize a module as part of a third party library.
known-third-party=enchant

# Couples of modules and preferred modules, separated by a comma.
preferred-modules=


[CLASSES]

# Warn about protected attribute access inside special methods
check-protected-access-in-special-methods=no

# List of method names used to declare (i.e. assign) instance attributes.
defining-attr-methods=__init__,
                      __new__,
                      setUp,
                      __post_init__

# List of member names, which should be excluded from the protected access
# warning.
exclude-protected=_asdict,
                  _fields,
                  _replace,
                  _source,
                  _make

# List of valid names for the first argument in a class method.
valid-classmethod-first-arg=cls

# List of valid names for the first argument in a metaclass class method.
valid-metaclass-classmethod-first-arg=cls


[EXCEPTIONS]

# Exceptions that will emit a warning when caught.
overgeneral-exceptions=BaseException,
                       Exception


[STRING]

# This flag controls whether inconsistent-quotes generates a warning when the
# character used as a quote delimiter is used inconsistently within a module.
check-quote-consistency=no

# This flag controls whether the implicit-str-concat should generate a warning
# on implicit string concatenation in sequences defined over several lines.
check-str-concat-over-line-jumps=no


[VARIABLES]

# List of additional names supposed to be defined in builtins. Remember that
# you should avoid defining new builtins when possible.
additional-builtins=

# Tells whether unused global variables should be treated as a violation.
allow-global-unused-variables=yes

# List of names allowed to shadow builtins
allowed-redefined-builtins=

# List of strings which can identify a callback function by name. A callback
# name must start or end with one of those strings.
callbacks=cb_,
          _cb

# A regular expression matching the name of dummy variables (i.e. expected to
# not be used).
dummy-variables-rgx=_+$|(_[a-zA-Z0-9_]*[a-zA-Z0-9]+?$)|dummy|^ignored_|^unused_

# Argument names that match this expression will be ignored. Default to name
# with leading underscore.
ignored-argument-names=_.*|^ignored_|^unused_

# Tells whether we should check for unused import in __init__ files.
init-import=no

# List of qualified module names which can have objects that can redefine
# builtins.
redefining-builtins-modules=six.moves,past.builtins,future.builtins,builtins,io


[DESIGN]

# List of regular expressions of class ancestor names to ignore when counting
# public methods (see R0903)
exclude-too-few-public-methods=

# List of qualified class names to ignore when counting class parents (see
# R0901)
ignored-parents=

# Maximum number of arguments for function / method.
max-args=5

# Maximum number of attributes for a class (see R0902).
max-attributes=7

# Maximum number of boolean expressions in an if statement (see R0916).
max-bool-expr=5

# Maximum number of branch for function / method body.
max-branches=12

# Maximum number of locals for function / method body.
max-locals=15

# Maximum number of parents for a class (see R0901).
max-parents=7

# Maximum number of public methods for a class (see R0904).
max-public-methods=20

# Maximum number of return / yield for function / method body.
max-returns=6

# Maximum number of statements in function / method body.
max-statements=50

# Minimum number of public methods for a class (see R0903).
min-public-methods=2


[REFACTORING]

# Maximum number of nested blocks for function / method body
max-nested-blocks=5

# Complete name of functions that never returns. When checking for
# inconsistent-return-statements if a never returning function is called then
# it will be considered as an explicit return statement and no message will be
# printed.
never-returning-functions=sys.exit


[TYPECHECK]

# List of decorators that produce context managers, such as
# contextlib.contextmanager. Add to this list to register other decorators that
# produce valid context managers.
contextmanager-decorators=contextlib.contextmanager

# List of members which are set dynamically and missed by pylint inference
# system, and so shouldn't trigger E1101 when accessed. Python regular
# expressions are accepted.
generated-members=

# Tells whether to warn about missing members when the owner of the attribute
# is inferred to be None.
ignore-none=yes

# This flag controls whether pylint should warn about no-member and similar
# checks whenever an opaque object is returned when inferring. The inference
# can return multiple potential results while evaluating a Python object, but
# some branches might not be evaluated, which results in partial inference. In
# that case, it might be useful to still emit no-member and other checks for
# the rest of the inferred objects.
ignore-on-opaque-inference=yes

# List of symbolic message names to ignore for Mixin members.
ignored-checks-for-mixins=no-member,
                          not-async-context-manager,
                          not-context-manager,
                          attribute-defined-outside-init

# List of class names for which member attributes should not be checked (useful
# for classes with dynamically set attributes). This supports the use of
# qualified names.
ignored-classes=optparse.Values,thread._local,_thread._local

# Show a hint with possible names when a member name was not found. The aspect
# of finding the hint is based on edit distance.
missing-member-hint=yes

# The minimum edit distance a name should have in order to be considered a
# similar match for a missing member name.
missing-member-hint-distance=1

# The total number of similar names that should be taken in consideration when
# showing a hint for a missing member.
missing-member-max-choices=1

# Regex pattern to define which classes are considered mixins.
mixin-class-rgx=.*[Mm]ixin

# List of decorators that change the signature of a decorated function.
signature-mutators=


[SIMILARITIES]

# Comments are removed from the similarity computation
ignore-comments=yes

# Docstrings are removed from the similarity computation
ignore-docstrings=yes

# Imports are removed from the similarity computation
ignore-imports=no

# Signatures are removed from the similarity computation
ignore-signatures=yes

# Minimum lines number of a similarity.
min-similarity-lines=4


[FORMAT]

# Expected format of line ending, e.g. empty (any line ending), LF or CRLF.
expected-line-ending-format=

# Regexp for a line that is allowed to be longer than the limit.
ignore-long-lines=^\s*(# )?<?https?://\S+>?$

# Number of spaces of indent required inside a hanging or continued line.
indent-after-paren=4

# String used as indentation unit. This is usually "    " (4 spaces) or "\t" (1
# tab).
indent-string='    '

# Maximum number of characters on a single line.
max-line-length=100

# Maximum number of lines in a module.
max-module-lines=1000

# Allow the body of a class to be on the same line as the declaration if body
# contains single statement.
single-line-class-stmt=no

# Allow the body of an if to be on the same line as the test if there is no
# else.
single-line-if-stmt=no


[SPELLING]

# Limits count of emitted suggestions for spelling mistakes.
max-spelling-suggestions=4

# Spelling dictionary name. Available dictionaries: none. To make it work,
# install the 'python-enchant' package.
spelling-dict=

# List of comma separated words that should be considered directives if they
# appear at the beginning of a comment and should not be checked.
spelling-ignore-comment-directives=fmt: on,fmt: off,noqa:,noqa,nosec,isort:skip,mypy:

# List of comma separated words that should not be checked.
spelling-ignore-words=

# A path to a file that contains the private dictionary; one word per line.
spelling-private-dict-file=

# Tells whether to store unknown words to the private dictionary (see the
# --spelling-private-dict-file option) instead of raising a message.
spelling-store-unknown-words=no
```

### Command used

```shell
pylint --score=no --enable-all-extensions pod-cmd
```

### Pylint output

```shell
Exception on node <ImportFrom l.10 at 0x7fbb1d8c8ca0> in file '/home/michael/bin/pod-cmd'
Traceback (most recent call last):
  File "/usr/local/lib/python3.10/dist-packages/pylint/utils/ast_walker.py", line 90, in walk
    callback(astroid)
  File "/usr/local/lib/python3.10/dist-packages/pylint/extensions/private_import.py", line 69, in visit_importfrom
    private_names = self._get_type_annotation_names(node, names)
  File "/usr/local/lib/python3.10/dist-packages/pylint/extensions/private_import.py", line 121, in _get_type_annotation_names
    self._populate_type_annotations(node.root(), self.all_used_type_annotations)
  File "/usr/local/lib/python3.10/dist-packages/pylint/extensions/private_import.py", line 170, in _populate_type_annotations
    ] = self._assignments_call_private_name(name_assignments, private_name)
  File "/usr/local/lib/python3.10/dist-packages/pylint/extensions/private_import.py", line 237, in _assignments_call_private_name
    current_attribute = current_attribute.expr
AttributeError: 'Name' object has no attribute 'expr'
************* Module pod-cmd
pod-cmd:1:0: F0002: pod-cmd: Fatal error while checking 'pod-cmd'. Please open an issue in our bug tracker so we address this. There is a pre-filled template that you can use in '/local/tmp/pylint.d/pylint-crash-2022-06-04-08.txt'. (astroid-error)
```
```


### Expected behavior

No crash

### Pylint version

```shell
pylint 2.14.0
astroid 2.11.5
Python 3.10.4 (main, Apr  2 2022, 09:04:19) [GCC 11.2.0]
```

### OS / Environment

```
$ cat /etc/lsb--release
DISTRIB_ID=Ubuntu
DISTRIB_RELEASE=22.04
DISTRIB_CODENAME=jammy
DISTRIB_DESCRIPTION="Ubuntu 22.04 LTS"
$
```

### Additional dependencies

apturl==0.5.2
arrow==0.14.7
astroid==2.11.5
bcrypt==3.2.0
blessings==1.7
blinker==1.4
bpython==0.22.1
Brlapi==0.8.3
certifi==2020.6.20
cffi==1.15.0
chardet==4.0.0
charset-normalizer==2.0.12
chrome-gnome-shell==0.0.0
click==8.0.3
colorama==0.4.4
command-not-found==0.3
cryptography==3.4.8
cupshelpers==1.0
curtsies==0.3.10
cwcwidth==0.1.6
cycler==0.11.0
dbus-python==1.2.18
defer==1.0.6
Deprecated==1.2.13
dill==0.3.4
distro==1.7.0
distro-info===1.1build1
duplicity==0.8.21
fasteners==0.14.1
flake8==4.0.1
fonttools==4.33.3
future==0.18.2
greenlet==1.1.2
httplib2==0.20.2
ics==0.7
idna==3.3
importlib-metadata==4.6.4
iso8601==1.0.2
isort==5.10.1
jeepney==0.7.1
keyring==23.5.0
kiwisolver==1.4.2
language-selector==0.1
launchpadlib==1.10.16
lazr.restfulclient==0.14.4
lazr.uri==1.0.6
lazy-object-proxy==1.7.1
lockfile==0.12.2
louis==3.20.0
macaroonbakery==1.3.1
matplotlib==3.5.2
mccabe==0.6.1
meld==3.20.4
monotonic==1.6
more-itertools==8.10.0
mypy==0.960
mypy-extensions==0.4.3
netifaces==0.11.0
numpy==1.22.3
oauthlib==3.2.0
olefile==0.46
packaging==21.3
paramiko==2.9.3
pexpect==4.8.0
Pillow==9.0.1
platformdirs==2.5.2
protobuf==3.12.4
ptyprocess==0.7.0
pycairo==1.20.1
pycodestyle==2.8.0
pycparser==2.21
pycups==2.0.1
pyflakes==2.4.0
PyGithub==1.55
Pygments==2.12.0
PyGObject==3.42.0
PyJWT==2.3.0
pylint==2.14.0
pymacaroons==0.13.0
PyNaCl==1.5.0
pyparsing==2.4.7
PyQt5==5.15.6
PyQt5-sip==12.9.1
PyQt5-stubs==5.15.6.0
pyRFC3339==1.1
pyserial==3.5
python-apt==2.3.0+ubuntu2
python-dateutil==2.8.2
python-debian===0.1.43ubuntu1
python-twitter==3.5
pytz==2022.1
pyxdg==0.27
PyYAML==5.4.1
reportlab==3.6.8
requests==2.27.1
requests-oauthlib==1.3.1
SecretStorage==3.3.1
six==1.16.0
ssh-import-id==5.11
stagger==1.0.0
systemd-python==234
TatSu==5.8.0
tomli==2.0.1
tomlkit==0.11.0
tqdm==4.64.0
tqdm-stubs==0.2.0
typeforce==0.2.0
types-python-dateutil==2.8.17
types-requests==2.27.29
types-urllib3==1.26.15
types-xmltodict==0.12.1
typing_extensions==4.2.0
ubuntu-advantage-tools==27.8
ubuntu-drivers-common==0.0.0
ufw==0.36.1
unattended-upgrades==0.1
urllib3==1.26.9
usb-creator==0.3.7
wadllib==1.3.6
wrapt==1.14.1
xdg==5
xkit==0.0.0
XlsxWriter==3.0.3
xmltodict==0.13.0
zipp==1.0.0

**Comment by jacobtylerwalls** [comment]
Duplicate of #6624
